backend/src/MiddleWare/OAuth.js

```diff
@@ -0,0 +1,24 @@
+require('dotenv').config();
+const jwt = require('jsonwebtoken');
+
+module.exports = function OAuth(req, res, next) {
+  const authorizationHeader = req.headers.authorization;
+  if (!authorizationHeader) {
+    res.status(415).send('Must include authorization info');
+    return;
+  }
+
+  if (!authorizationHeader.startsWith('Bearer ')) {
+    res.status(415).send('Must use Bearer OAuth token!');
+    return;
+  }
+
+  const token = authorizationHeader.replace('Bearer ', '');
+  try {
+    const userId = jwt.verify(token, process.env.JWT_PRIVATE_KEY);
+    res.locals.userId = userId;
```

Comment:
it is still the request part 
```suggestion
    req.userId = userId;
```